service/hostapi/hostapi.go

```diff
@@ -58,13 +58,17 @@ func StartUp() {
 		}
 		break
 	}
+	var rancherClient *rclient.RancherClient
 	for {
-		rancherClient, err := util.GetRancherClient()
-		if err != nil {
-			logrus.Errorf("Failed to get rancher client for host-api startup: %v", err)
-			time.Sleep(time.Duration(5) * time.Second)
-			continue
+		var err error
+		rancherClient, err = util.GetRancherClient()
+		if err == nil {
+			break
 		}
+		logrus.Errorf("Failed to get rancher client for host-api startup: %v", err)
+		time.Sleep(time.Duration(5) * time.Second)
+	}
+	for {
```

Comment:
Why do we need this change? how does it cause a nil pointer reference?

Reply:
@LLParse I am not quite sure. Do you recall anything?

Reply:
I don't recall. It would be best to remove the changes and rediscover the root problem (if any)